python/triton/compiler/code_generator.py

```diff
@@ -1546,9 +1546,16 @@ def ret(self, node: ast.Call):
 
 def ast_to_ttir(fn, src, context, options, codegen_fns, module_map, module=None):
     arg_types = [None] * len(fn.arg_names)
-    for k, v in src.signature.items():
-        idx = fn.arg_names.index(k)
-        arg_types[idx] = str_to_ty(v)
+    const_iter = iter(src.constants.items())
+    kc, vc = next(const_iter, (None, None))
+
+    for i, (ks, v) in enumerate(src.signature.items()):
+        idx = fn.arg_names.index(ks)
+        cexpr = None
+        if kc is not None and kc[0] == i:
```

Comment:
This is not correct; it does not support nested tuples. Doing this properly probably requires recursion: https://github.com/triton-lang/triton/pull/7442#issuecomment-3058516025

Reply:
#8248